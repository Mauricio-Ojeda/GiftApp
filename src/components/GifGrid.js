import React  from 'react';
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

     
    const {data: images, loading} = useFetchGifs( category );
    
    return (
        <>
            <h3>{ category }</h3>

            { loading && <h2 className=" animate__hinge "> Cargando... 🐱‍🏍</h2> }
            
            <div className="card-grid ">
                {
                    images.map( img =>(
                    
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />

                    ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
