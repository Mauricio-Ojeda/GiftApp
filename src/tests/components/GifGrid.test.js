
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en GifGrid', () => {
    
    const category = 'hola Prisma';

    test('crear snapshot de GifGrid', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category = { category } /> );

        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe mostrar items cuando cargan las imagenes de useFetchGifs', () => {
        
        // sirve para falsear simular algo

        const gifs = [{
            id: 'abc',
            url: 'https://localhost.gato.jpg',
            title: 'cualquiera'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category = { category } /> );

        expect(wrapper.find('h2').exists()).toBe(true);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
    
    
})
