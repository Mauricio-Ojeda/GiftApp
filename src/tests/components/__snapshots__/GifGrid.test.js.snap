// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGrid crear snapshot de GifGrid 1`] = `
<Fragment>
  <h3>
    hola Prisma
  </h3>
  <h2
    className=" animate__hinge "
  >
     Cargando... 🐱‍🏍
  </h2>
  <div
    className="card-grid "
  />
</Fragment>
`;

exports[`pruebas en GifGrid debe mostrar items cuando cargan las imagenes de useFetchGifs 1`] = `
<Fragment>
  <h3>
    hola Prisma
  </h3>
  <h2
    className=" animate__hinge "
  >
     Cargando... 🐱‍🏍
  </h2>
  <div
    className="card-grid "
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquiera"
      url="https://localhost.gato.jpg"
    />
  </div>
</Fragment>
`;
